### PR TITLE
ARROW-4539: [Java] Fix child vector count for lists.

### PR DESCRIPTION
**Pull request by praveenbingo**
- Child vector count was not set correctly for lists. Fixed to use the right count.

### CODE REVIEW

**Comment by praveenbingo** [comment]
@siddharthteotia - could you please review..

File: java/vector/src/test/java/org/apache/arrow/vector/TestVarCharListVector.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package org.apache.arrow.vector;
```
Comment (xhochy) [inline java/vector/src/test/java/org/apache/arrow/vector/TestVarCharListVector.java]: This files needs an Apache license header.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: java/vector/src/test/java/org/apache/arrow/vector/TestVarCharListVector.java (hunk 1 above)
Comment (praveenbingo) [reply java/vector/src/test/java/org/apache/arrow/vector/TestVarCharListVector.java]: @xhochy yup missed it..added them now.

File: java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java [hunk 2]
```diff
@@ -744,7 +744,7 @@ public void setValueCount(int valueCount) {
     }
     /* valueCount for the data vector is the current end offset */
     final int childValueCount = (valueCount == 0) ? 0 :
-            offsetBuffer.getInt(valueCount * OFFSET_WIDTH);
+            offsetBuffer.getInt(lastSet * OFFSET_WIDTH);
```
Comment (siddharthteotia) [inline java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java]: So were we always setting the value count for child vector incorrectly? I am not clear what the problem is. Is there a special case we have run into?

Comment (mikepigott) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java]: Hi!  I ran into this over on #3294, when I tried to convert a string array from a JDBC field into a VarChar ListVector.  If one of the values in the JDBC string array was `null`, I couldn't read the value after it from the VarChar ListVector (the following value came back as an empty string).

@praveenbingo was kind enough to put together this PR to fix the issue.

Comment (praveenbingo) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java]: Yup, @siddharthteotia - we were not correctly handling lists that could have null values (setting child vector count using valuecount will not handle cases where valuecount < lastSet i.e. presence of nulls).

It does not affect List of FixedWidthVectors and only VariableWidthVectors. 

Please let me know if you would need more information.

Comment (siddharthteotia) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java]: Thanks @mikepigott  and @praveenbingo

**Review by siddharthteotia — APPROVED** [review]
+1

**Pull request merged**